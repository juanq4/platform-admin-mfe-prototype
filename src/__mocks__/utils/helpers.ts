import { isEmpty } from "@q4/nimbus-ui";

export function mockWindowMethods(): Window {
  return Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
}

export function mockClipboardWrite(): void {
  const orginalClipboard = { ...navigator.clipboard };

  beforeAll(() => {
    if (!isEmpty(navigator.clipboard)) return;

    Object.defineProperty(navigator, "clipboard", {
      value: {
        writeText: () => Promise.resolve(),
      },
    });
  });

  afterAll(() => {
    Object.assign(navigator.clipboard, {
      ...orginalClipboard,
    });
  });
}
