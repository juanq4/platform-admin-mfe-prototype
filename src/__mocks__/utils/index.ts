export * from "./helpers";
export * from "./permissions";
export * from "./wrappers";
