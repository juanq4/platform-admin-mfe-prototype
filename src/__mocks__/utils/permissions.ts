import { Permission } from "@q4/platform-definitions";

export const MockQ4AdminPermissions = [
  Permission.ReadUsers,
  Permission.ManageUsers,
  Permission.ReadOrgs,
  Permission.ManageOrgs,
];

export const MockQ4SupportPermissions = [Permission.ReadUsers, Permission.ReadOrgs, Permission.ManageOrgs];

export const MockCorporateAdminPermissions = [Permission.ReadUsers, Permission.ManageUsers];

export const MockCorporateSupportPermissions: Permission[] = [];
