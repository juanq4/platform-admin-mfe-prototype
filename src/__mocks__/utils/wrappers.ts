import { NimbusConfig, NimbusProviderImplementationMock } from "../contexts";

export const createAppWrapper = (): void => {
  const body = global.document.querySelector("body");
  const rootElement = global.document.createElement("div");

  rootElement.setAttribute("id", "root");
  body.appendChild(rootElement);

  const nimbusConfigSpy = jest.spyOn(NimbusConfig, "ConfigProvider");
  nimbusConfigSpy.mockImplementation(NimbusProviderImplementationMock);
};

export const destroyAppWrapper = (): void => {
  const body = global.document.querySelector("body");
  const rootElement = global.document.getElementById("root");

  if (rootElement) {
    body.removeChild(rootElement);
  }
};

export const getAppWrapper = (): HTMLElement => {
  return global.document.getElementById("root");
};
