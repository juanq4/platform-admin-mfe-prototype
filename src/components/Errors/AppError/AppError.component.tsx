import { Anchor, AnchorTheme, Fallback, Text } from "@q4/nimbus-ui";
import { Email } from "../../../configurations";

export const AppError = (): JSX.Element => {
  return (
    <Fallback
      email={Email.Support}
      placeholderContentProps={{
        subtitle: (
          <div>
            <div>
              <Text>There was a problem loading Q4 Platform.</Text>
            </div>
            <div>
              <Text>
                Please refresh the page or contact{" "}
                <Anchor theme={AnchorTheme.Primary} url={`mailto:${Email.Support}`}>
                  {Email.Support}
                </Anchor>
                .
              </Text>
            </div>
          </div>
        ),
      }}
    />
  );
};
