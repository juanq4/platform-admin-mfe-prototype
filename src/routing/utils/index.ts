export * from "./api";
export * from "./s3checks";
export * from "./versionManager";
