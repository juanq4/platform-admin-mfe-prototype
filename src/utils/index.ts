export * from "./account";
export * from "./api";
export * from "./entitlement";
export * from "./error";
export * from "./feature";
export * from "./logout";
export * from "./organization";
export * from "./permission";
export * from "./role";
export * from "./route";
export * from "./subscription";
export * from "./user";
export * from "./environment";
