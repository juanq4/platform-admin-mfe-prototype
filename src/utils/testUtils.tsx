/* eslint-disable import/export */
import { MockedProvider as ApolloMockedProvider } from "@apollo/client/testing";
import { ThemeProvider } from "@emotion/react";
import { StyleGuide, ConfigProvider as NimbusConfig } from "@q4/nimbus-ui";
import type { RenderOptions, RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import theme from "../styles/theme";

// https://testing-library.com/docs/react-testing-library/setup/
const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
  styleGuide = StyleGuide.V2,
): RenderResult =>
  render(ui, {
    // Common middleware to all tests
    wrapper: ({ children }) => (
      <NimbusConfig styleGuide={styleGuide}>
        <ApolloMockedProvider>
          <ThemeProvider theme={theme}>{children}</ThemeProvider>
        </ApolloMockedProvider>
      </NimbusConfig>
    ),
    ...options,
  });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
