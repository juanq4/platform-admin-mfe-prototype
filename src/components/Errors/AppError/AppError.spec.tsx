import { render } from "../../../utils/testUtils";
import { AppError } from "./AppError.component";

describe("AppError", () => {
  it("7386736: renders component", () => {
    const { container } = render(<AppError />);
    expect(container).toMatchSnapshot();
  });
});
