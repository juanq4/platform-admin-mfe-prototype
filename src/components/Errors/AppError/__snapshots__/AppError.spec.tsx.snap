// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppError 7386736: renders component 1`] = `
.emotion-0 {
  background-color: #f7fafc;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  padding: 64px 96px;
}

.emotion-1 {
  background: transparent;
  -webkit-transition: background 0.3s ease-in-out;
  transition: background 0.3s ease-in-out;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: unset;
  -ms-flex-direction: unset;
  flex-direction: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: auto;
  padding: 24px;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

@media only screen and (max-width: 768px) {
  .emotion-1 {
    padding: 16px;
  }
}

.emotion-1 .nui-placeholder-content_image {
  height: 235px;
  max-width: 312px;
}

.emotion-1 .nui-placeholder-content_title {
  align: center;
  font-size: 18px;
  font-weight: 600;
  line-height: 24px;
  margin-left: auto;
  margin-right: auto;
  max-width: 500px;
}

.emotion-1 .nui-placeholder-content_subtitle {
  align: center;
  margin-left: auto;
  margin-right: auto;
  max-width: 500px;
}

.emotion-3 {
  -webkit-transition: color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out;
  text-align: center;
}

.emotion-5 {
  display: inline-block;
  width: 100%;
  max-width: 190px;
  margin-bottom: 16px;
}

.emotion-7 {
  font-family: 'Open Sans',sans-serif;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
  font-style: normal;
  font-weight: 600;
  font-size: 20px;
  line-height: 28px;
  margin-block-start: 0;
  margin-block-end: 0;
  font-weight: 600;
}

@media only screen and (max-width: 768px) {
  .emotion-7 {
    font-size: 18px;
    line-height: 28px;
  }
}

.emotion-9 {
  font-family: 'Open Sans',sans-serif;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 18px;
  margin-block-start: 0;
  margin-block-end: 0;
}

@media only screen and (max-width: 768px) {
  .emotion-9 {
    font-size: 10px;
    line-height: 16px;
  }
}

.emotion-10 {
  font-family: 'Open Sans',sans-serif;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
}

@media only screen and (max-width: 768px) {
  .emotion-10 {
    font-size: 12px;
    line-height: 12px;
  }
}

.emotion-12 {
  font-size: 14px;
  line-height: 16px;
  background-color: transparent;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  border-radius: 4px;
  padding: 0 2px;
}

.emotion-12:focus {
  background-color: transparent;
  outline: 1px solid #0f5ca3;
}

.emotion-12:hover {
  background-color: #e6f2fd;
  color: #0a3d6d;
}

.emotion-12,
.emotion-12:visited {
  color: #0f5ca3;
}

@media only screen and (max-width: 768px) {
  .emotion-12 {
    padding: 0 2px;
    font-size: 12px;
    line-height: 12px;
  }
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin: 16px 0 24px;
}

.emotion-15 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  background-color: #0f5ca3;
  color: #ffffff;
  border-color: transparent;
  font-family: inherit;
  height: 40px;
  padding: 12px 16px;
  border-width: 1px;
  border-style: solid;
  font-size: 14px;
  line-height: 16px;
  text-transform: none;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  cursor: pointer!important;
  border-radius: 4px;
}

.emotion-15:hover {
  background: #0a3d6d;
  color: #ffffff;
}

.emotion-15:hover [data-type="nui-tooltip"] {
  opacity: 1;
  visibility: visible;
}

.emotion-15:focus {
  background: #0a3d6d;
  border-color: #4ea1ef;
  box-shadow: 0px 0px 3px #4ea1ef;
}

.emotion-15:active {
  background: #051f36;
}

.emotion-15:disabled {
  pointer-events: none;
  cursor: default;
  background: #f7fafc;
  color: #61707c;
  border-color: transparent;
}

@media only screen and (max-width: 768px) {
  .emotion-15 {
    height: 40px;
    padding: 12px;
  }
}

.emotion-16 {
  display: inline-block;
  cursor: pointer;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      height="auto"
    >
      <div
        class="emotion-2 emotion-3"
      >
        <img
          alt="Oops! Something went wrong."
          class="emotion-4 emotion-5"
          src="data:image/svg+xml,%3c%3fxml version='1.0' encoding='UTF-8'%3f%3e%3csvg width='312px' height='233px' viewBox='0 0 312 233' version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'%3e %3c!-- Generator: Sketch 62 (91390) - https://sketch.com --%3e %3ctitle%3eGroup 3%3c/title%3e %3cdesc%3eCreated with Sketch.%3c/desc%3e %3cg id='Page-1' stroke='none' stroke-width='1' fill='none' fill-rule='evenodd'%3e %3cg id='Error-State' transform='translate(-480.000000%2c -123.000000)'%3e %3cg id='Group-3' transform='translate(473.000000%2c 117.000000)'%3e %3cellipse id='Oval-Copy' fill='%234696E0' fill-rule='nonzero' opacity='0.1' cx='165.5' cy='229' rx='130.5' ry='10'%3e%3c/ellipse%3e %3cg id='Group-2-Copy' transform='translate(51.500000%2c 113.500000) rotate(-330.000000) translate(-51.500000%2c -113.500000) translate(9.000000%2c 86.000000)' fill='%23297AC5' fill-rule='nonzero' opacity='0.1'%3e %3cpath d='M62%2c52.2187697 C57.9600199%2c51.4645913 53.7840152%2c51.911746 50.0020771%2c53.5034731 C45.4119389%2c55.5110691 40.173101%2c55.4982368 35.5931225%2c53.468179 C30.1698686%2c51.1221268 23.9761417%2c51.2434173 18.6515117%2c53.7999431 C17.1468464%2c54.5820215 15.4728901%2c54.9937085 13.7722132%2c55 C6.90399261%2c55 1.18762982%2c48.1740719 0%2c39.1670308 C1.37244052%2c38.1803547 2.53855004%2c36.9411385 3.43411032%2c35.5176263 C7.46203554%2c29.115286 13.7006693%2c25 20.7048234%2c25 C27.7089776%2c25 33.868913%2c29.0658743 37.875375%2c35.3976265 C40.3246218%2c39.3011407 44.6697053%2c41.6509694 49.3224094%2c41.5882025 L49.5012693%2c41.5882025 C54.9243019%2c41.5811437 59.6605123%2c45.908194 62%2c52.2187697 Z' id='Path'%3e%3c/path%3e %3cpath d='M82.1363405%2c24.006237 L71.1432739%2c30.9234075 L77.815612%2c18.881626 C75.9246159%2c17.3836046 73.5839093%2c16.5548377 71.1645234%2c16.5266949 L70.9874443%2c16.5266949 C70.2236455%2c16.5377805 69.4603698%2c16.4813203 68.7066663%2c16.3579835 L64.9950897%2c18.7410331 L66.5958842%2c15.8588787 C63.9545544%2c14.9428636 61.673344%2c13.2202742 60.0793757%2c10.9381273 L53.4070375%2c15.1559142 L57.6073523%2c7.49360136 C53.7045303%2c2.84700615 48.4488245%2c0 42.6689647%2c0 C35.7345497%2c0 29.5580328%2c4.09125328 25.5702129%2c10.4671411 C23.165678%2c14.3621205 18.8371989%2c16.6737846 14.2371546%2c16.5196653 L13.861747%2c16.5196653 C6.20484945%2c16.5196653 0%2c25.1309802 0%2c35.7598031 C0%2c46.3886261 6.20484945%2c55 13.861747%2c55 C15.546576%2c54.9954979 17.2048325%2c54.5828594 18.6924631%2c53.7978717 C23.9651209%2c51.2541607 30.0971682%2c51.1359415 35.4653895%2c53.474508 C39.999766%2c55.4961782 45.1864417%2c55.5089575 49.7308767%2c53.5096563 C55.0582142%2c51.2148898 61.1273111%2c51.3303268 66.3621399%2c53.8259903 C67.835701%2c54.5913459 69.4730406%2c54.9939717 71.1361907%2c55 C78.7930883%2c55 84.9979377%2c46.3815964 84.9979377%2c35.7598031 C85.0513264%2c31.6658913 84.0673331%2c27.6242919 82.1363405%2c24.006237 Z' id='Path'%3e%3c/path%3e %3c/g%3e %3cg id='Group-2-Copy-2' transform='translate(276.500000%2c 151.500000) scale(-1%2c 1) translate(-276.500000%2c -151.500000) translate(234.000000%2c 124.000000)' fill='%23297AC5' fill-rule='nonzero' opacity='0.1'%3e %3cpath d='M62%2c52.2187697 C57.9600199%2c51.4645913 53.7840152%2c51.911746 50.0020771%2c53.5034731 C45.4119389%2c55.5110691 40.173101%2c55.4982368 35.5931225%2c53.468179 C30.1698686%2c51.1221268 23.9761417%2c51.2434173 18.6515117%2c53.7999431 C17.1468464%2c54.5820215 15.4728901%2c54.9937085 13.7722132%2c55 C6.90399261%2c55 1.18762982%2c48.1740719 0%2c39.1670308 C1.37244052%2c38.1803547 2.53855004%2c36.9411385 3.43411032%2c35.5176263 C7.46203554%2c29.115286 13.7006693%2c25 20.7048234%2c25 C27.7089776%2c25 33.868913%2c29.0658743 37.875375%2c35.3976265 C40.3246218%2c39.3011407 44.6697053%2c41.6509694 49.3224094%2c41.5882025 L49.5012693%2c41.5882025 C54.9243019%2c41.5811437 59.6605123%2c45.908194 62%2c52.2187697 Z' id='Path'%3e%3c/path%3e %3cpath d='M82.1363405%2c24.006237 L71.1432739%2c30.9234075 L77.815612%2c18.881626 C75.9246159%2c17.3836046 73.5839093%2c16.5548377 71.1645234%2c16.5266949 L70.9874443%2c16.5266949 C70.2236455%2c16.5377805 69.4603698%2c16.4813203 68.7066663%2c16.3579835 L64.9950897%2c18.7410331 L66.5958842%2c15.8588787 C63.9545544%2c14.9428636 61.673344%2c13.2202742 60.0793757%2c10.9381273 L53.4070375%2c15.1559142 L57.6073523%2c7.49360136 C53.7045303%2c2.84700615 48.4488245%2c0 42.6689647%2c0 C35.7345497%2c0 29.5580328%2c4.09125328 25.5702129%2c10.4671411 C23.165678%2c14.3621205 18.8371989%2c16.6737846 14.2371546%2c16.5196653 L13.861747%2c16.5196653 C6.20484945%2c16.5196653 0%2c25.1309802 0%2c35.7598031 C0%2c46.3886261 6.20484945%2c55 13.861747%2c55 C15.546576%2c54.9954979 17.2048325%2c54.5828594 18.6924631%2c53.7978717 C23.9651209%2c51.2541607 30.0971682%2c51.1359415 35.4653895%2c53.474508 C39.999766%2c55.4961782 45.1864417%2c55.5089575 49.7308767%2c53.5096563 C55.0582142%2c51.2148898 61.1273111%2c51.3303268 66.3621399%2c53.8259903 C67.835701%2c54.5913459 69.4730406%2c54.9939717 71.1361907%2c55 C78.7930883%2c55 84.9979377%2c46.3815964 84.9979377%2c35.7598031 C85.0513264%2c31.6658913 84.0673331%2c27.6242919 82.1363405%2c24.006237 Z' id='Path'%3e%3c/path%3e %3c/g%3e %3cpath d='M288%2c46.2705882 C284.0243%2c45.5390207 279.913853%2c45.9734454 276.194197%2c47.5183193 C271.681745%2c49.4675911 266.527619%2c49.4675911 262.015167%2c47.5183193 C256.677752%2c45.2503625 250.582325%2c45.3679652 245.342235%2c47.84 C243.86364%2c48.597038 242.217663%2c48.9950864 240.5455%2c49 C233.787834%2c49 228.166502%2c42.4006723 227%2c33.6957983 C228.347397%2c32.7384 229.494146%2c31.540774 230.378833%2c30.1670588 C234.340917%2c23.9771429 240.434883%2c20 247.363501%2c20 C254.29212%2c20 260.32575%2c23.8991597 264.287834%2c30.050084 C266.695054%2c33.8066869 270.948433%2c36.0750986 275.510386%2c36.0352941 L275.691395%2c36.0352941 C281.021101%2c36.0060504 285.677052%2c40.1878992 288%2c46.2705882 Z' id='Path' fill='%23297AC5' fill-rule='nonzero' opacity='0.1' transform='translate(257.500000%2c 34.500000) rotate(-240.000000) translate(-257.500000%2c -34.500000) '%3e%3c/path%3e %3cg id='Group-2' transform='translate(59.000000%2c 61.000000)'%3e %3cg id='Group'%3e %3cpolygon id='Path-Copy' fill='%23CCCCCC' fill-rule='nonzero' points='134 168 79 167.303951 79.1635894 165.914894 81.821146 143 130.123239 143 133.675908 165.914894 133.947528 167.653495'%3e%3c/polygon%3e %3cpath d='M212%2c4.56379007 C212.008479%2c2.05409851 209.973348%2c0.0118000705 207.452201%2c0 L4.54782293%2c0 C3.33596989%2c0.00566543754 2.17601358%2c0.490322531 1.32313799%2c1.34734939 C0.470262407%2c2.20437625 -0.00566724542%2c3.36356761 0%2c4.56990775 L0%2c127 L212%2c127 L212%2c4.56379007 Z' id='Path-Copy-2' fill='%23545B62' fill-rule='nonzero'%3e%3c/path%3e %3cpath d='M0%2c125 L0%2c139.440986 C0%2c141.95886 2.03610762%2c144 4.54777365%2c144 L207.452226%2c144 C209.963892%2c144 212%2c141.95886 212%2c139.440986 L212%2c125 L0%2c125 Z' id='Path-Copy-3' fill='%23DDDDDD' fill-rule='nonzero'%3e%3c/path%3e %3crect id='Rectangle-Copy-2' fill='%23EEEEEE' fill-rule='nonzero' x='8' y='7' width='195' height='110'%3e%3c/rect%3e %3crect id='Rectangle' fill='white' x='8' y='15' width='195' height='102'%3e%3c/rect%3e %3ccircle id='Oval' fill='%23939BA0' cx='106.5' cy='134.5' r='3.5'%3e%3c/circle%3e %3cg id='Group-6' transform='translate(41.000000%2c 24.000000)' fill-rule='nonzero'%3e %3cg id='Group-5'%3e %3cpath d='M56.7160662%2c23.8461278 C56.1876054%2c21.7264307 55.3070059%2c19.6067336 54.2500843%2c17.663861 L56.363759%2c12.7180137 L48.9659819%2c5.47581382 L44.0341867%2c7.59551095 C42.0966656%2c6.53574691 40.1591445%2c5.65246933 37.8694848%2c5.12250279 L36.1081174%2c0 L25.7160662%2c0 L23.7785451%2c5.12250279 C21.6648704%2c5.65246933 19.5511957%2c6.53557786 17.6138432%2c7.59551095 L12.6818795%2c5.47581382 L5.46025601%2c12.7180137 L7.57393068%2c17.663861 C6.51717763%2c19.6069026 5.63640961%2c21.5499442 5.1079488%2c23.8461278 L0%2c25.7891695 L0%2c36.2108305 L5.1079488%2c38.1538722 C5.63640961%2c40.2735693 6.51700906%2c42.3932664 7.57393068%2c44.336139 L5.46025601%2c49.2819863 L12.8580331%2c56.5241862 L17.7898283%2c54.4044891 C19.7273493%2c55.4642531 21.6648704%2c56.3475307 23.9545301%2c56.8774972 L25.8920512%2c62 L36.1079488%2c62 L38.0454699%2c56.8774972 C40.1591445%2c56.3475307 42.2728192%2c55.4644221 44.2101717%2c54.4044891 L49.1419669%2c56.5241862 L56.539744%2c49.2819863 L54.4260693%2c44.336139 C55.4828224%2c42.3930974 56.3635904%2c40.4500558 56.8920512%2c38.1538722 L62%2c36.2108305 L62%2c25.7891695 L56.7160662%2c23.8461278 Z M31%2c43.704918 C23.941731%2c43.704918 18.295082%2c38.058269 18.295082%2c31 C18.295082%2c23.941731 23.941731%2c18.295082 31%2c18.295082 C38.058269%2c18.295082 43.704918%2c23.941731 43.704918%2c31 C43.704918%2c38.058269 38.058269%2c43.704918 31%2c43.704918 Z' id='Shape-Copy' fill='%234696E0'%3e%3c/path%3e %3cpath d='M39.7155048%2c39.7154606 C34.7073849%2c44.7235707 26.6584802%2c44.7235707 21.6503603%2c39.7154606 L17%2c44.3658119 C24.5122654%2c51.8780627 36.8537709%2c51.8780627 44.3658651%2c44.3658119 C51.8779594%2c36.8535611 51.8781305%2c24.690959 44.3658651%2c17 L39.7155048%2c21.6503513 C44.7237958%2c26.6584614 44.7237958%2c34.7073505 39.7155048%2c39.7154606 Z' id='Path' fill='%231F6BB0'%3e%3c/path%3e %3cpath d='M17.6341881%2c17.6341881 C10.1219373%2c25.1464389 10.1219373%2c37.309041 17.6341881%2c45 L22.2845394%2c40.3496487 C17.2764293%2c35.3415386 17.2764293%2c27.2926495 22.2845394%2c22.2845394 C27.2926495%2c17.2764293 35.3415386%2c17.2764293 40.3496487%2c22.2845394 L45%2c17.6341881 C37.4879204%2c10.1219373 25.1464389%2c10.1219373 17.6341881%2c17.6341881 Z' id='Path' fill='%231F6BB0'%3e%3c/path%3e %3c/g%3e %3cpath d='M93.4213821%2c67.6917882 C93.6792686%2c66.1447162 93.6792686%2c64.8552838 93.4213821%2c63.3082118 L96%2c60.7295939 L93.4213821%2c54.2831725 L89.8112183%2c54.2831725 C89.0375589%2c53.2516266 87.7483734%2c52.2203276 86.7168275%2c51.1887817 L86.7168275%2c47.5786179 L80.2704061%2c45 L77.6917882%2c47.5786179 C76.1447162%2c47.3207314 74.8552838%2c47.3207314 73.3082118%2c47.5786179 L70.7295939%2c45 L64.2831725%2c47.5786179 L64.2831725%2c51.1887817 C63.2516266%2c51.9624411 62.2203276%2c53.2516266 61.1887817%2c54.2831725 L57.5786179%2c54.2831725 L55%2c60.7295939 L57.5786179%2c63.3082118 C57.3207314%2c64.8552838 57.3207314%2c66.1447162 57.5786179%2c67.6917882 L55%2c70.2704061 L57.5786179%2c76.7168275 L61.1887817%2c76.7168275 C61.9624411%2c77.7483734 63.2516266%2c78.7796724 64.2831725%2c79.8112183 L64.2831725%2c83.4213821 L70.7295939%2c86 L73.3082118%2c83.4213821 C74.8552838%2c83.6792686 76.1447162%2c83.6792686 77.6917882%2c83.4213821 L80.2704061%2c86 L86.7168275%2c83.4213821 L86.7168275%2c79.8112183 C87.7483734%2c79.0375589 88.7796724%2c77.7483734 89.8112183%2c76.7168275 L93.4213821%2c76.7168275 L96%2c70.2704061 L93.4213821%2c67.6917882 Z M72.3645743%2c73.3383384 C67.9228672%2c71.5094556 65.8326667%2c66.5451664 67.6615608%2c62.3645412 C69.490455%2c57.9228615 74.4547748%2c55.8326739 78.6354257%2c57.6615568 C83.0771328%2c59.4904396 85.1673333%2c64.4547288 83.3384392%2c68.6353539 C81.7706013%2c73.0770337 76.8062815%2c75.1674713 72.3645743%2c73.3383384 Z' id='Shape-Copy-2' fill='%23CCCCCC'%3e%3c/path%3e %3cpath d='M122.068161%2c32.6917882 C122.332338%2c31.1447162 122.332338%2c29.8552838 122.068161%2c28.3082118 L124.709672%2c25.7295939 L122.068161%2c19.2831725 L118.369945%2c19.2831725 C117.577416%2c18.2516266 116.256787%2c17.2203276 115.200081%2c16.1887817 L115.200081%2c12.5786179 L108.59643%2c10 L105.954919%2c12.5786179 C104.370113%2c12.3207314 103.049232%2c12.3207314 101.464426%2c12.5786179 L98.822915%2c10 L92.2192638%2c12.5786179 L92.2192638%2c16.1887817 C91.1625583%2c16.9624411 90.1061056%2c18.2516266 89.0494%2c19.2831725 L85.3511835%2c19.2831725 L82.7096725%2c25.7295939 L85.3511835%2c28.3082118 C85.0870071%2c29.8552838 85.0870071%2c31.1447162 85.3511835%2c32.6917882 L82.7096725%2c35.2704061 L85.3511835%2c41.7168275 L89.0494%2c41.7168275 C89.8419292%2c42.7483734 91.1625583%2c43.7796724 92.2192638%2c44.8112183 L92.2192638%2c48.4213821 L98.822915%2c51 L101.464426%2c48.4213821 C103.049232%2c48.6792686 104.370113%2c48.6792686 105.954919%2c48.4213821 L108.59643%2c51 L115.200081%2c48.4213821 L115.200081%2c44.8112183 C116.256787%2c44.0375589 117.313239%2c42.7483734 118.369945%2c41.7168275 L122.068161%2c41.7168275 L124.709672%2c35.2704061 L122.068161%2c32.6917882 Z M100.38981%2c39.2994171 C95.6868259%2c37.362953 93.4736725%2c32.1066468 95.4101487%2c27.6801025 C97.3466248%2c22.9771474 102.602963%2c20.7640077 107.029535%2c22.7004719 C111.732519%2c24.6369361 113.945672%2c29.8932423 112.009196%2c34.3197865 C110.349133%2c39.0227416 105.092794%2c41.2361461 100.38981%2c39.2994171 Z' id='Shape-Copy-3' fill-opacity='0.6' fill='%234696E0'%3e%3c/path%3e %3c/g%3e %3c/g%3e %3crect id='Rectangle-Copy' fill='%23CCCCCC' fill-rule='nonzero' x='69' y='167' width='73' height='1'%3e%3c/rect%3e %3c/g%3e %3c/g%3e %3c/g%3e %3c/g%3e%3c/svg%3e"
        />
        <h3
          class="emotion-6 emotion-7"
        >
          Oops! Something went wrong.
        </h3>
        <p
          class="emotion-8 emotion-9"
        >
          <div>
            <div>
              <span
                class="emotion-10"
              >
                There was a problem loading Q4 Platform.
              </span>
            </div>
            <div>
              <span
                class="emotion-10"
              >
                Please refresh the page or contact
                 
                <a
                  class="emotion-12"
                  href="mailto:q4platform-support@q4inc.com"
                  tabindex="0"
                >
                  q4platform-support@q4inc.com
                </a>
                .
              </span>
            </div>
          </div>
        </p>
        <div
          class="emotion-13 emotion-14"
        >
          <button
            class="emotion-15"
            type="button"
          >
            <span
              class="emotion-16"
            >
              Refresh
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
